raft/server.cc

```diff
@@ -645,7 +656,7 @@ future<entry_id> server_impl::add_entry_on_leader(command cmd, seastar::abort_so
         try {
             memory_permit = co_await _fsm->wait_for_memory_permit(as, log::memory_usage_of(cmd, _config.max_command_size));
         } catch (semaphore_aborted&) {
-            throw request_aborted();
+            throw request_aborted(format("Semaphore aborted while waiting for memory availability for adding entry on leader in term: {}, on server: {}", t, _id));
```

Comment:
nit: I'd also print `_fsm->get_current_term()` (as it might have changed and be different than `t` by the moment we abort).